Update for game version v21.5.0

megalon opened the pull request:
This PR was created by comparing game versions `20.1.0` (before the Alt Costumes update) and `21.5.0` (Garfield Update) decompiled with ILSpy.

I created a local repo and committed the contents of 20.1.0, then added in the contents for 21.5.0 and manually checked the differences to see what changed.

I may have missed something, it's difficult to be sure. I was able to convert Garfield's moveset `char_orb` using this new version.
Before the changes in this PR, NASB_Parser failed to convert his moveset.

megalon committed:
Add new attributes to FSMovement

megalon committed:
Update InputValidator

megalon committed:
Update Jump

megalon committed:
Update SAConfigHitbox

megalon committed:
Update SAFindLastHorizontalInput

megalon committed:
Update StateAction

megalon committed:
Add SAMapAnimationSimple

I am not sure that I did this right

megalon committed:
Add KnockbackAltJump

megalon committed:
Add DelayedJump

sc2ad reviewed (changes requested):
Normally I'm not such a stickler for formatting, but the braces and inconsistent indentation bothers me perhaps more than it should. In addition, the missing accessibility modifiers are probably something you don't intend for-- you probably want them to be public? Instead of default internal.

megalon committed:
Fix accessibility modifiers

megalon committed:
Fix formatting

megalon committed:
Fix HurtBone

It looks like forceZ0 is just being ignored now. They haven't removed it from the class though.

megalon commented:
Ok, I think I've fixed the accessibility modifiers and the formatting.
Also fixed the forceZ in the HurtBone class. It seems they are just ignoring it now.

sc2ad commented on NASB_Parser/StateActions/SAConfigHitbox.cs:
```diff
@@ -52,7 +52,7 @@ public override void Write(BulkSerializeWriter writer)
             writer.Write(TID);
             writer.Write(1);
             writer.Write(Hitbox);
-            writer.Write(ForceZ0);
```
What changed here, are you sure this is correct?

megalon replied:
Yeah, between the two versions that I compared.
Here is a screenshot from the repo I'm using to compare.
The latest update, 21.5.0, is on the right.

![image](https://user-images.githubusercontent.com/27714637/145506124-ab2d739f-fd4a-421e-a08a-09408cdffd05.png)

megalon replied:
The other change in that file is simply hiding the toggle for that boolean in their "EditDraw" function, so I assume it's an unused value now.
![image](https://user-images.githubusercontent.com/27714637/145506304-a6f3b460-0623-403a-a633-39d6dae065b5.png)

megalon committed:
Add new Jump TIDs to Read method

megalon committed:
Fix Write method for SAMapAnimationSimple

megalon committed:
Fix line endings

megalon commented:
I did another pass on the changes today and noticed a few mistakes in my PR, and fixed with the commits above.
Also, to fix the test case, do I need to simply update the char_apple.txt?

megalon committed:
Set all ForceZ0 to 0 in test moveset

This is to account for the changes in SAConfigHitbox and HurtBone
